RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT.cc

```diff
@@ -0,0 +1,778 @@
+  
+  if (T0>1./betamax){
+    return betamax/pow(coolingFactor_, int(std::log(T0*betamax)/std::log(coolingFactor_))-1 );
+  }else{
+    // ensure at least one annealing step
+    return betamax/coolingFactor_;
+  }
+}
+
+bool DAClusterizerInZT::split( double beta,
+                               vector<track_t> & tks,
+                               vector<vertex_t> & y,
+                               double threshold ) const {
+  // split only critical vertices (Tc >~ T=1/beta   <==>   beta*Tc>~1)
+  // an update must have been made just before doing this (same beta, no merging)
+  // returns true if at least one cluster was split
+
+  constexpr double epsilon=1e-3;      // split all single vertices by 10 um 
+  bool split=false;
+  
+  // avoid left-right biases by splitting highest Tc first
+  
+  std::vector<std::pair<double, unsigned int> > critical;
+  for(unsigned int ik=0; ik<y.size(); ik++){
+    if (beta*y[ik].Tc > 1.){
+      critical.push_back( make_pair(y[ik].Tc, ik));
+    }
+  }
+  std::stable_sort(critical.begin(), critical.end(), std::greater<std::pair<double, unsigned int> >() );
+
+  for(unsigned int ic=0; ic<critical.size(); ic++){
+    unsigned int ik=critical[ic].second;
+    // estimate subcluster positions and weight
+    double p1=0, z1=0, t1=0, w1=0;
+    double p2=0, z2=0, t2=0, w2=0;
+    //double sumpi=0;
+    for(unsigned int i=0; i<tks.size(); i++){
+      if(tks[i].Z>0){
+	//sumpi+=tks[i].pi;
+	double p=y[ik].pk * exp(-beta*Eik(tks[i],y[ik])) / tks[i].Z*tks[i].pi;
+	double w=p/(tks[i].dz2 * tks[i].dt2);
+	if(tks[i].z < y[ik].z){
+	  p1+=p; z1+=w*tks[i].z; t1+=w*tks[i].t; w1+=w;
+	}else{
+	  p2+=p; z2+=w*tks[i].z; t2+=w*tks[i].t; w2+=w;
+	}
+      }
+    }
+    if(w1>0){  z1=z1/w1; t1=t1/w1;} else{ z1=y[ik].z-epsilon; t1=y[ik].t-epsilon; }
+    if(w2>0){  z2=z2/w2; t2=t2/w2;} else{ z2=y[ik].z+epsilon; t2=y[ik].t+epsilon;}
+
+    // reduce split size if there is not enough room
+    if( ( ik   > 0       ) && ( y[ik-1].z>=z1 ) ){ z1=0.5*(y[ik].z+y[ik-1].z); t1=0.5*(y[ik].t+y[ik-1].t); }
+    if( ( ik+1 < y.size()) && ( y[ik+1].z<=z2 ) ){ z2=0.5*(y[ik].z+y[ik+1].z); t2=0.5*(y[ik].t+y[ik+1].t); }
+
+    // split if the new subclusters are significantly separated
+    if( (z2-z1)>epsilon || std::abs(t2-t1) > epsilon){
+      split=true;
+      vertex_t vnew;
+      vnew.pk = p1*y[ik].pk/(p1+p2);
+      y[ik].pk= p2*y[ik].pk/(p1+p2);
+      vnew.z  = z1;
+      vnew.t  = t1;
+      y[ik].z = z2;
+      y[ik].t = t2;
+      y.insert(y.begin()+ik, vnew);
+
+     // adjust remaining pointers
+      for(unsigned int jc=ic; jc<critical.size(); jc++){
+	if (critical[jc].second>ik) {critical[jc].second++;}
+      }
+    }
+  }
+
+  //  stable_sort(y.begin(), y.end(), clusterLessZ);
+  return split;
+}
+
+
+
+
+
+void DAClusterizerInZT::splitAll( vector<vertex_t> & y ) const {
+
+
+  constexpr double epsilon=1e-3;      // split all single vertices by 10 um 
```

Comment:
looks like these 1e-3 are related to the other ones in this file.
Change to a constexpr in the DAClusterizerInZT members